feat(add-inbox): implement custom backend and socket URL configuration fixes NV-6580

merrcury opened the pull request:
### What changed? Why was the change needed?

- Added support for custom backend and socket URLs in the Novu Inbox component.
- Updated CLI to accept `backendUrl` and `socketUrl` as command line arguments.
- Enhanced component generation functions to utilize custom URLs or fallback to region-specific defaults.
- Updated documentation to reflect new configuration options.

### Screenshots

<!-- If the changes are visual, include screenshots or screencasts. -->

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Related enterprise PR

<!-- A link to a dependent pull request  -->

### Special notes for your reviewer

<!-- Specific instructions or considerations you want to highlight for the reviewer. -->

</details>

merrcury committed:
feat(add-inbox): implement custom backend and socket URL configuration

- Added support for custom backend and socket URLs in the Novu Inbox component.
- Updated CLI to accept `backendUrl` and `socketUrl` as command line arguments.
- Enhanced component generation functions to utilize custom URLs or fallback to region-specific defaults.
- Updated documentation to reflect new configuration options.

merrcury committed:
fix(add-inbox): update region handling when custom URLs are provided

- Adjusted region parameter logic to prioritize custom backend and socket URLs.
- Updated CLI documentation to clarify that the region option is optional when using custom URLs.

iampearceman committed:
Merge branch 'next' into fix/add-inbox-selfhosted

iampearceman commented on packages/add-inbox/src/generators/frameworks/react.ts:
```diff
@@ -32,7 +37,30 @@ function isReactVersionModern(version: string): boolean {
+  const finalBackendUrl = backendUrl || (region === 'eu' ? 'https://eu.api.novu.co' : null);
+  const finalSocketUrl = socketUrl || (region === 'eu' ? 'https://eu.ws.novu.co' : null);
+
+  // Build URL props string
```
Missing string escaping for URL values.

Unlike the Next.js implementation, this code doesn't escape the URL strings before injecting them into the generated code. This could lead to issues if URLs contain special characters like quotes or backslashes.

```
   let urlProps = '';
   if (finalBackendUrl || finalSocketUrl) {
     const props = [];
     if (finalBackendUrl) {
-      props.push(`backendUrl="${finalBackendUrl}"`);
+      props.push(`backendUrl="${finalBackendUrl.replace(/\\/g, '\\\\').replace(/"/g, '\\"')}"`);
     }
     if (finalSocketUrl) {
-      props.push(`socketUrl="${finalSocketUrl}"`);
+      props.push(`socketUrl="${finalSocketUrl.replace(/\\/g, '\\\\').replace(/"/g, '\\"')}"`);
     }
     urlProps = `\n    ${props.join(' ')}`;
   }
```

merrcury replied:
@iampearceman I have fixed this. But Now I have a doubt. In staging snipped says wss  for socketurl and eu it says https. Please let me know which one is correct

iampearceman replied:
I think it was a mistake. Based on the docs [here](https://docs.novu.co/platform/sdks/react#usage) it should be `ws`.

merrcury committed:
fix(add-inbox): enforce protocol validation for backend and socket URLs

- Updated URL validation functions to enforce HTTP/HTTPS for backend URLs and WS/WSS for socket URLs.
- Adjusted default socket URL in Next.js generator to use WSS protocol.
- Enhanced React generator to escape special characters in URLs for better handling.

merrcury committed:
fix(add-inbox): update socket URL protocol in documentation

- Changed socket URL in the example from "https" to "wss" for consistency with protocol validation.
- Updated documentation to reflect the correct WSS protocol for default region-specific URLs.
